plugins/rikki/heroeslounge/http/Match.php

```diff
@@ -78,25 +83,36 @@ public function games($id)
         return $retVal;
     }
 
-    public function getTodaysMatches($date, $tz1 = TimezoneHelper::DEFAULT_TIMEZONE, $tz2 = "")
+    public function getTodaysMatches($tz1 = TimezoneHelper::DEFAULT_TIMEZONE, $tz2 = "")
     {
         if ($tz2) {
             $timezone = $tz1 . '/' . $tz2;
         } else {
             $timezone = $tz1;
         }
 
-        // get today in a given timezone
-        if ($date == 'today') {
-            $start = Carbon::today($timezone);
+        $start = Carbon::today($timezone);
+
+        // now convert back to default/server timezone for db query
+        $start->setTime(0,0,0)->setTimezone(TimezoneHelper::DEFAULT_TIMEZONE);
+        $end = $start->copy()->addDay();
+        return MatchModel::whereBetween('wbp', [$start, $end])->where('is_played',false)->get();
+    }
+
+    public function getMatchesByDate($date, $tz1 = TimezoneHelper::DEFAULT_TIMEZONE, $tz2 = "")
+    {
+        if ($tz2) {
+            $timezone = $tz1 . '/' . $tz2;
         } else {
-            $start = Carbon::createFromFormat('Y-m-d', $date, $timezone);
+            $timezone = $tz1;
         }
 
+        $start = Carbon::createFromFormat('Y-m-d', $date, $timezone);
+
         // now convert back to default/server timezone for db query
         $start->setTime(0,0,0)->setTimezone(TimezoneHelper::DEFAULT_TIMEZONE);
         $end = $start->copy()->addDay();
-        return MatchModel::whereBetween('wbp', [$start, $end])->where('is_played',false)->get();
+        return MatchModel::whereBetween('wbp', [$start, $end])->get();
```

Comment:
I used one method for both to avoid all this copy paste.  You can still have one method, just make it private and pass different args in from the calling methods that the api hooks into

Reply:
Did some refactoring, not entirely sure if I'm happy wit the outcome, but I think it's okay